store/sqlstore/store.go

```diff
@@ -403,6 +428,42 @@ func (ss *SqlStore) TotalMasterDbConnections() int {
 	return ss.GetMaster().Db.Stats().OpenConnections
 }
 
+// ReplicaLagAbs queries all the replica databases to get the absolute replica lag value
+// and updates the Prometheus metric with it.
+func (ss *SqlStore) ReplicaLagAbs() {
+	for i, item := range ss.settings.ReplicaLagSettings {
+		if *item.QueryAbsoluteLag == "" {
+			continue
+		}
+		var binDiff int
+		var node string
+		err := ss.replicaLagHandles[i].QueryRow(*item.QueryAbsoluteLag).Scan(&node, &binDiff)
+		if err != nil {
+			mlog.Warn("ReplicaLagAbs Query returned an error", mlog.Err(err))
+			return
+		}
+		ss.metrics.SetReplicaLagAbsolute(node, float64(binDiff))
+	}
+}
+
+// ReplicaLagAbs queries all the replica databases to get the time-based replica lag value
+// and updates the Prometheus metric with it.
+func (ss *SqlStore) ReplicaLagTime() {
+	for i, item := range ss.settings.ReplicaLagSettings {
+		if *item.QueryTimeLag == "" {
+			continue
+		}
+		var timeDiff int
```

Comment:
Are we confident that this metric will always fit a int? I am wondering if there are cases where this data could be a floating point number on the db side. In any case maybe we should mention the datatype requirement in the release note.

Reply:
Good point. I will update.